src/components/ha-bar.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+
+import {
+  getValueInPercentage,
+  normalize,
+  roundWithOneDecimal,
+} from "../util/calculate";
+
+@customElement("ha-bar")
+export class HaBar extends LitElement {
+  @property({ type: Number }) public min = 0;
+
+  @property({ type: Number }) public max = 100;
+
+  @property({ type: Number }) public value!: number;
+
+  @property({ type: Number }) public target?: number; // Number between 0 and 100
+
+  protected render(): TemplateResult {
+    const valuePrecentage = roundWithOneDecimal(
+      getValueInPercentage(
+        normalize(this.value, this.min, this.max),
+        this.min,
+        this.max
+      )
+    );
+
+    return html`
+      <slot></slot>
```

Comment:
What is the slot for?

Reply:
A description, a name, something
The example uses that for the "Sample value" text

Reply:
I feel like the component should be just that. Feels weird that the slot is for this one use case of the text to the left of the bar. But I guess you can always just use it or not use it 👍

Reply:
I agree with Zack, this should not be in the component

Reply:
In that case, I'll drop all text, and just have the bar, so can the implementation do everything else.